### PR TITLE
Configurable properties

### PR DESCRIPTION
**Pull request by Werkov**
I'm posting this for review. The new config options aren't deliberately documented in the conf-file since they shouldn't be officially necessary. Also mind it's still speculative if this is needed at all, so _de_priotize accordingly.

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -71,17 +108,50 @@ static int parse_list(struct str_list *rl, char *s, const char *delim) {
...
+static int parse_kill_mode(void *target, char *v) {
+	char **p = target;
+	/* Accept "none" only for the time being */
+	if (strcmp(KILL_MODE_NONE, v)) {
+		return -EINVAL;
+	}
+
+	free(*p);
+	*p = v;
+}
```
Comment (nick-wang) [inline src/config.c:132]: Need return value due to `static int` type.

File: src/config.c [hunk 2]
```diff
@@ -71,17 +108,50 @@ static int parse_list(struct str_list *rl, char *s, const char *delim) {
...
+		return 0;
+	}
+
+	errno = 0;
+	limit = strtoull(v, NULL, 10);
+	if (errno)
+		return -errno;
+
+	*p = limit;
+}
```
Comment (nick-wang) [inline src/config.c:151]: Same here, need return value.

File: src/config.c [hunk 3]
```diff
@@ -71,17 +108,50 @@ static int parse_list(struct str_list *rl, char *s, const char *delim) {
...
+
+static int parse_str(void *target, char *v) {
+	char **p = target;
+	free(*p);
+	*p = v;
+}
```
Comment (nick-wang) [inline src/config.c:158]: Same here, need return value.

File: src/config.c (hunk 2 above)
Comment (nick-wang) [inline src/config.c:151]: Need to free `v` when return with 0.

**Review by nick-wang — CHANGES_REQUESTED** [review]
Great change, except minor issues could be enhanced.

File: src/config.c (hunk 2 above)
Comment (Werkov) [reply src/config.c:151]: Good catch.

**Comment by Werkov** [comment]
Thanks for the review, I've force-pushed fixes.

**Review by nick-wang — APPROVED** [review]

**Comment by nick-wang** [comment]
Looks perfect after the changes, ready to merge!

**Pull request merged**